### PR TITLE
CMSIS-NN: Fix static code issues

### PR DESCRIPTION
**Pull request by mansnils**
* Resolve BAD_SHIFT, DIVIDE_BY_ZERO and some CERT INT34-C issues.
* Do not build q7 files, i.e. only s8 files.

### CODE REVIEW

**Comment by mansnils** [comment]
@felix-johnny can you review this?

File: CMSIS/NN/Source/SoftmaxFunctions/arm_softmax_s8.c [hunk 1]
```diff
@@ -21,8 +21,8 @@
  * Title:        arm_softmax_s8.c
  * Description:  S8 softmax function
  *
- * $Date:        09. October 2020
- * $Revision:    V.2.0.1
+ * $Date:        01. March 2021
+ * $Revision:    V.2.0.2
```
Comment (felix-johnny) [inline CMSIS/NN/Source/SoftmaxFunctions/arm_softmax_s8.c:25]: This would be a 3.0.0 change as it is a change in the API..

**Review by felix-johnny — CHANGES_REQUESTED** [review]

File: CMSIS/NN/Source/ConvolutionFunctions/CMakeLists.txt [hunk 2]
```diff
@@ -4,7 +4,7 @@ project(CMSISNNConvolutions)
 
 include(configLib)
 
-file(GLOB SRC "./*_*.c")
+file(GLOB SRC "./*_s8.c")
```
[the remark this answers is not in the document]
Comment (mansnils) [reply CMSIS/NN/Source/ConvolutionFunctions/CMakeLists.txt:7]: I also noticed that in my next ongoing PR, I will add a fix for it here.

File: CMSIS/NN/Source/SoftmaxFunctions/arm_softmax_s8.c (hunk 1 above)
Comment (mansnils) [reply CMSIS/NN/Source/SoftmaxFunctions/arm_softmax_s8.c:25]: Ok. I suppose the same goes for arm_nnfunctions.h

**Review by felix-johnny — APPROVED** [review]

**Comment by felix-johnny** [comment]
Thanks for the changes @mansnils and apologies for the delay in review..

**Pull request merged**